cmd_test.go

```diff
@@ -0,0 +1,73 @@
+package orlop
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.ketch.com/lib/orlop/v2/config"
+	"go.uber.org/fx"
+)
+
+func TestRun(t *testing.T) {
+	os.Setenv("TEST_CONFIG_EMBEDDED", "true")
+	os.Setenv("TEST_CONFIG_REQ", "imhere")
+	os.Setenv("TEST_CONFIG_SLICED", "\"a\",\"b\",\"c\"")
+	os.Setenv("TEST_CONFIG_CUSTOM", "1m")
+	os.Setenv("TEST_CONFIG_MAP", "[\"a=b\",\"c=d\"]")
+	os.Setenv("TEST_CONFIG_HEX_ENCODED", "0102030405060708090A0B0C0D0E0F")
+	os.Setenv("TEST_CONFIG_BASE_64_ENCODED", "AQIDBAUGBwgJCgsMDQ4P")
+	os.Setenv("TEST_CONFIG_PTR", "123")
+
+	var cfg TestConfig
+
+	var module = fx.Options(
+		fx.Provide(
+			func(provider config.Provider) (*TestConfig, error) {
+				c, err := provider.Get(context.Background(), "config")
+				if err != nil {
+					return nil, err
+				}
+				return c.(*TestConfig), nil
+			},
+		),
+		fx.Invoke(
+			func(provider config.Provider) {
+				provider.Register(context.Background(), "config", &cfg)
+			},
+		),
+		fx.Invoke(
+			func(lifecycle fx.Lifecycle, config *TestConfig) {
+				lifecycle.Append(
+					fx.Hook{
+						OnStart: func(context.Context) error {
+							return nil
+						},
+						OnStop: func(context.Context) error {
+							return nil
+						},
+					},
+				)
+			},
+		),
+	)
+
+	//TODO: This is not ideal
+	go Run("test", module)
+
+	time.Sleep(1 * time.Second)
+
```

Comment:
Not sure about how to write a better test using `fxtest`. Any suggestions? 

In the meantime, will keep tinkering around.